code/datums/storage.dm

```diff
@@ -0,0 +1,291 @@
+/datum/storage
+	var/atom/parent
+
+	var/max_slots
+	var/max_specific_storage
+	var/max_total_storage
+
+	var/list/is_using = list()
+
+	var/locked = FALSE
+	var/attack_hand_interact = TRUE
+	var/allow_big_nesting = FALSE
+
+	var/atom/movable/screen/storage/boxes //storage display object
+	var/atom/movable/screen/close/closer //close button object
+
+	//Screen variables: Do not mess with these vars unless you know what you're doing. They're not defines so storage that isn't in the same location can be supported in the future.
+	var/screen_max_columns = 7 //These two determine maximum screen sizes.
+	var/screen_max_rows = INFINITY
+	var/screen_pixel_x = 16 //These two are pixel values for screen loc of boxes and closer
+	var/screen_pixel_y = 16
+	var/screen_start_x = 4 //These two are where the storage starts being rendered, screen_loc wise.
+	var/screen_start_y = 2
+
+
+
+/datum/storage/New(atom/parent, max_slots, max_specific_storage, max_total_storage)
+	boxes = new(null, src)
+	closer = new(null, src)
+
+	src.parent = parent
+	src.max_slots = max_slots
+	src.max_specific_storage = max_specific_storage
+	src.max_total_storage = max_total_storage
+
+	orient_to_hud()
+
+/datum/storage/Destroy()
+	. = ..()
+
+	parent = null
+	boxes = null
+	closer = null
+	is_using.Cut()
+
+
+
+/datum/storage/proc/reset_item(obj/item/thing)
+	thing.layer = initial(thing.layer)
+	thing.plane = initial(thing.plane)
+	thing.mouse_opacity = initial(thing.mouse_opacity)
+	if(thing.maptext)
+		thing.maptext = ""
+
+
+
+/datum/storage/proc/attempt_insert(obj/item/to_insert, mob/user)
+	message_admins("ran")
+	if(!isitem(to_insert))
+		return FALSE
+
+	if(to_insert == parent)
+		return FALSE
+
+	if(to_insert.loc == parent)
+		return FALSE
```

Comment:
do you only care if it or its loc is the parent?

Reply:
yes i know i can combine them i probably will

Reply:
what happens if our loc's loc is our parent?